src/jshint.js

```diff
@@ -3413,88 +3409,31 @@ var JSHINT = (function() {
     });
   }
 
-  var conststatement = stmt("const", function(context) {
+  function variableStatement(type, statement, context) {
```

Comment:
If @jugglinmike is alright with your approach, doing it all in one function is fine too --- but in my experience that leads to more problems